### DISCUSSION TITLE
.NET 6 and Oqtane 3.0

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#1726] [General]
The following pull request contains the changes required to upgrade Oqtane to .NET 6.  

https://github.com/oqtane/oqtane.framework/pull/1725

** Please note that this is still a work in progress and therefore should not be considered production ready. The code base has been upgraded to use the latest RC2 packages and all compatibility issues have been addressed to ensure the framework compiles without any errors or warnings. New installations have been successfully tested on SQL Server, LocalDB, and SQLite.

Module and Theme templates still need to be upgraded to .NET 6, and third party modules need to be tested for compatibility. The Oqtane 3.0 release is not expected to be published until the week of Nov 8th when the official .NET 6 packages are made available. 

.NET 6 is a significant step forward for Oqtane as it is an LTS release for both Blazor Server and Blazor WebAssembly which means that it will be supported until November 2024 ( whereas .NET 5 support ends in May 2022 ).

**Comment by curlyfro** [comment]
this is great!  thanks Shawn :)

**Reply by markagama** [reply]
great work @sbwalker

**Comment by artmedia** [comment]
great job

**Comment by wbonekamp** [comment]
Can we expect Oqtane version 3.0 this week? or is it rescheduled?

**Reply by hishamco** [reply]
Oqtane is already using .NET RC https://github.com/oqtane/oqtane.framework/commit/29bd31f609d194f558b3d22fad03b115308c00ea I expect the release will be published today after VS 2022 launch, @sbwalker can you confirm when Oqtane 3.0.0 will be released?

**Reply by sbwalker** [reply]
@wbonekamp Oqtane 3.0 will be released this week. The dev branch is already using the RC but all of the package references will need to be updated to the official .NET 6 packages once they are available. I was planning to do the official release announcement in my .NET Conf presentation on Thursday.

**Reply by sbwalker** [reply]
I have updated the package references to the official .NET 6 packages however there are still some packages which appear to not have been released yet:

```<PackageReference Include="Microsoft.Data.SqlClient" Version="4.0.0-preview3.21293.2" />```

```<PackageReference Include="System.ComponentModel.Annotations" Version="6.0.0-preview.4.21253.7" />```

Newer versions are not available yet...

**Reply by sbwalker** [reply]
SqlClient is apparently not going to be released until ~ Nov 20

**Comment by sbwalker** [comment]
I just published the 3.0.0 release packages. A few items to note:

- some of the package references do not reflect the official .NET "6.0.0" version. This is because Microsoft has not yet released official versions of some packages yet. A good example is SqlClient which is not expected to be released until later this month. We will update Oqtane with official package references in a subsequent release once they become available. 

- I successfully upgraded an Oqtane 2.3.1 release running on .NET 5 to Oqtane 3.0.0 running on .NET 6. In order to do this I first needed to install the new Hosting Bundle for .NET 6. In addition, because I was using a local install of SQL Server I ran into an issue related to the breaking change by Microsoft in .NET 6 where they are now requiring encrypted database connections. So I had to manually update the connection string in my appsettings.json to include the "Encrypt=false;" qualifier. I am mentioning this because the error message which was displayed during upgrade was not intuitive and it took some time for me to track this down.

- I will be publishing a more detailed release blog announcement by the end of this week. Today I am focused on the .NET Conf presentation I need to provide this afternoon related to "Creating Database-Agnostic Applications with EF Core" - which obviously uses Oqtane as the demo.

**Comment by sbwalker** [comment]
FYI.... The official release blog is published here:

https://www.oqtane.org/Resources/Blog/PostId/551/announcing-oqtane-30-for-net-6

And the recording of my session from .NET Conf 2021 is here:

https://www.youtube.com/watch?v=lal5zVWU6YU&list=PLdo4fOcmZ0oVFtp9MDEBNbA2sSqYvXSXO&index=74

**Reply by curlyfro** [reply]
i didn't know you were one of the presenters... I'm watching now.  Thanks, Shaun for your contributions!

**Reply by curlyfro** [reply]
just watched your video.  i always knew dnn and oqtane had multi-tenant features but watching your presentation was provokingly awesome!  but I can't do anything with oqtane until 3.1 :(
